shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -490,6 +490,34 @@ - (void)dealloc {
   [super dealloc];
 }
 
+#pragma mark - Status Bar touch event handling
+
+// Standard iOS status bar height in pixels.
+constexpr CGFloat kStandardStatusBarHeight = 20.0;
+
+- (void)handleStatusBarTouches:(UIEvent *)event {
+  // If the status bar is double-height, don't handle status bar taps. iOS
+  // should open the app associated with the status bar.
+  CGRect statusBarFrame = [UIApplication sharedApplication].statusBarFrame;
+  if (statusBarFrame.size.height != kStandardStatusBarHeight) {
+    return;
+  }
+
+  // If we detect a touch in the status bar, synthesize a fake touch begin/end.
+  for (UITouch *touch in event.allTouches) {
+    if (touch.phase == UITouchPhaseBegan && touch.tapCount > 0) {
+      CGPoint windowLoc = [touch locationInView:nil];
+      CGPoint screenLoc = [touch.window convertPoint:windowLoc toWindow:nil];
+      if (CGRectContainsPoint(statusBarFrame, screenLoc)) {
+        NSSet *statusbarTouches = [NSSet setWithObject:touch];
+        [self dispatchTouches:statusbarTouches phase:UITouchPhaseBegan];
+        [self dispatchTouches:statusbarTouches phase:UITouchPhaseEnded];
```

Comment:
I can take a look at setting phase to UITouchPhaseEnded on the second event. We don't rely on it elsewhere, since we use the phase param from the second param, but probably worth doing for future-proofing.